apis/ql.d.ts

```diff
@@ -231,29 +231,32 @@ export class DELETE<T> extends ConstructedQuery<T> {
   DELETE: CQN.DELETE['DELETE']
 
 }
+// operator for qbe expression
+type QbeOp = '=' | '-=' | '+=' | '*=' | '/=' | '%='
 
 export interface UPDATE<T> extends Where<T>, And, ByKey {}
 export class UPDATE<T> extends ConstructedQuery<T> {
   private constructor();
 
   // cds-typer plural
-  // FIXME: this returned UPDATE<SingularInstanceType<T>> before. should UPDATE<Books>.entity(...) return Book or Books?
-  static entity<T extends ArrayConstructable> (entity: T, primaryKey?: PK): UPDATE<InstanceType<T>>
-
-  static entity (entity: EntityDescription, primaryKey?: PK): UPDATE<StaticAny>
-
-  static entity<T extends Constructable> (entity: T, primaryKey?: PK): UPDATE<T>
-
-  // currently no easy way to restrict T from being a primitive type
-  static entity<T> (entity: T, primaryKey?: PK): UPDATE<T>
+  static entity: (TaggedTemplateQueryPart<UPDATE<StaticAny>>)
+   // FIXME: this returned UPDATE<SingularInstanceType<T>> before. should UPDATE<Books>.entity(...) return Book or Books?
+    & (<T extends ArrayConstructable> (entity: T, primaryKey?: PK) => UPDATE<SingularInstanceType<T>>)
```

Comment:
so I guess you made a decision here, regarding the comment above. :) What was your reasoning here?

Reply:
The reasoning is simple. If we do not return the singular type, the type inference will result in the following
![image](https://github.com/user-attachments/assets/7d1be400-c248-4cb5-9092-b3887007f433)

Reply:
Okay, that makes sense! Guess the `FIXME` comment can be removed or adjusted to give the explanation you just provided